### ISSUE TITLE
"default" is missing in BindingFlags

### ISSUE THREAD

**Issue by gavr123456789** [StefanSalewski/gintro#173]
```nim
type
  BindingFlag* {.size: sizeof(cint), pure.} = enum
    bidirectional = 0
    syncCreate = 1
    invertBoolean = 2

  BindingFlags* {.size: sizeof(cint).} = set[BindingFlag]
  ```  
  
  https://docs.gtk.org/gobject/flags.BindingFlags.html  
  https://valadoc.org/gobject-2.0/GLib.BindingFlags.html  
    
  But there must be also G_BINDING_DEFAULT

**Comment by StefanSalewski** [comment]
Thanks for reporting. Default is the empty set {}, but we should have a named constant for this.

**Comment by StefanSalewski** [comment]
You may test with

```
nimble uninstall gintro
nimble install gintro@#head
```

This is the upcomming v0.9.4 which I intend to ship this evening.

I have done a test for the gtk3 examples compiled with ARC, and for sdt compiled with ARC. Compiles and runs fine for me. Maybe I will do a few more test later this day.

For your default flags, it should work now:

```
salewski@nuc ~/gintrotest/tests $  grep -C6 BindingFlagsDefault nim_gi/gobject.nim 
    syncCreate = 1
    invertBoolean = 2

  BindingFlags* {.size: sizeof(cint).} = set[BindingFlag]

const
  BindingFlagsDefault* = BindingFlags({})
proc default*(t: typedesc[BindingFlags]): BindingFlags = BindingFlags({})


$ cat ~/gintrotest/tests/t.nim 
import gintro/gobject

echo BindingFlagsDefault == BindingFlags.default


$ diff nim_gi/ ~/.nimble/pkgs/gintro-#v0.9.2/gintro/
Only in /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/: gimplgst.nim
diff nim_gi/gio.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gio.nim
1712,1715d1711
< const
<   IOStreamSpliceFlagsNone* = IOStreamSpliceFlags({})
< proc none*(t: typedesc[IOStreamSpliceFlags]): IOStreamSpliceFlags = IOStreamSpliceFlags({})
< 
1725,1728d1720
< const
<   FileMonitorFlagsNone* = FileMonitorFlags({})
< proc none*(t: typedesc[FileMonitorFlags]): FileMonitorFlags = FileMonitorFlags({})
< 
2048,2051d2039
< const
<   FileCopyFlagsNone* = FileCopyFlags({})
< proc none*(t: typedesc[FileCopyFlags]): FileCopyFlags = FileCopyFlags({})
< 
2059,2062d2046
< const
<   FileCreateFlagsNone* = FileCreateFlags({})
< proc none*(t: typedesc[FileCreateFlags]): FileCreateFlags = FileCreateFlags({})
< 
2082,2085d2065
< const
<   FileQueryInfoFlagsNone* = FileQueryInfoFlags({})
< proc none*(t: typedesc[FileQueryInfoFlags]): FileQueryInfoFlags = FileQueryInfoFlags({})
< 
3733,3736d3712
< const
<   MountUnmountFlagsNone* = MountUnmountFlags({})
< proc none*(t: typedesc[MountUnmountFlags]): MountUnmountFlags = MountUnmountFlags({})
< 
4789,4792d4764
< const
<   AppInfoCreateFlagsNone* = AppInfoCreateFlags({})
< proc none*(t: typedesc[AppInfoCreateFlags]): AppInfoCreateFlags = AppInfoCreateFlags({})
< 
7763,7766d7734
< const
<   DBusCallFlagsNone* = DBusCallFlags({})
< proc none*(t: typedesc[DBusCallFlags]): DBusCallFlags = DBusCallFlags({})
< 
8056,8059d8023
< const
<   DBusCapabilityFlagsNone* = DBusCapabilityFlags({})
< proc none*(t: typedesc[DBusCapabilityFlags]): DBusCapabilityFlags = DBusCapabilityFlags({})
< 
8080,8083d8043
< const
<   DBusConnectionFlagsNone* = DBusConnectionFlags({})
< proc none*(t: typedesc[DBusConnectionFlags]): DBusConnectionFlags = DBusConnectionFlags({})
< 
8639,8642d8598
< const
<   DBusSubtreeFlagsNone* = DBusSubtreeFlags({})
< proc none*(t: typedesc[DBusSubtreeFlags]): DBusSubtreeFlags = DBusSubtreeFlags({})
< 
9393,9396d9348
< const
<   DBusSendMessageFlagsNone* = DBusSendMessageFlags({})
< proc none*(t: typedesc[DBusSendMessageFlags]): DBusSendMessageFlags = DBusSendMessageFlags({})
< 
9445,9448d9396
< const
<   DBusMessageFlagsNone* = DBusMessageFlags({})
< proc none*(t: typedesc[DBusMessageFlags]): DBusMessageFlags = DBusMessageFlags({})
< 
9533,9536d9480
< const
<   DBusSignalFlagsNone* = DBusSignalFlags({})
< proc none*(t: typedesc[DBusSignalFlags]): DBusSignalFlags = DBusSignalFlags({})
< 
11100,11103d11043
< const
<   FileAttributeInfoFlagsNone* = FileAttributeInfoFlags({})
< proc none*(t: typedesc[FileAttributeInfoFlags]): FileAttributeInfoFlags = FileAttributeInfoFlags({})
< 
11302,11305d11241
< const
<   ApplicationFlagsFlagsNone* = ApplicationFlags({})
< proc flagsNone*(t: typedesc[ApplicationFlags]): ApplicationFlags = ApplicationFlags({})
< 
11762,11765d11697
< const
<   OutputStreamSpliceFlagsNone* = OutputStreamSpliceFlags({})
< proc none*(t: typedesc[OutputStreamSpliceFlags]): OutputStreamSpliceFlags = OutputStreamSpliceFlags({})
< 
12677,12680d12608
< const
<   DBusObjectManagerClientFlagsNone* = DBusObjectManagerClientFlags({})
< proc none*(t: typedesc[DBusObjectManagerClientFlags]): DBusObjectManagerClientFlags = DBusObjectManagerClientFlags({})
< 
12707,12710d12634
< const
<   DBusProxyFlagsNone* = DBusProxyFlags({})
< proc none*(t: typedesc[DBusProxyFlags]): DBusProxyFlags = DBusProxyFlags({})
< 
14351,14354d14274
< const
<   BusNameOwnerFlagsNone* = BusNameOwnerFlags({})
< proc none*(t: typedesc[BusNameOwnerFlags]): BusNameOwnerFlags = BusNameOwnerFlags({})
< 
14364,14367d14283
< const
<   BusNameWatcherFlagsNone* = BusNameWatcherFlags({})
< proc none*(t: typedesc[BusNameWatcherFlags]): BusNameWatcherFlags = BusNameWatcherFlags({})
< 
14736,14739d14651
< const
<   ConverterFlagsNone* = ConverterFlags({})
< proc none*(t: typedesc[ConverterFlags]): ConverterFlags = ConverterFlags({})
< 
15522,15525d15433
< const
<   DBusInterfaceSkeletonFlagsNone* = DBusInterfaceSkeletonFlags({})
< proc none*(t: typedesc[DBusInterfaceSkeletonFlags]): DBusInterfaceSkeletonFlags = DBusInterfaceSkeletonFlags({})
< 
16148,16151d16055
< const
<   DBusPropertyInfoFlagsNone* = DBusPropertyInfoFlags({})
< proc none*(t: typedesc[DBusPropertyInfoFlags]): DBusPropertyInfoFlags = DBusPropertyInfoFlags({})
< 
16408,16411d16311
< const
<   DBusServerFlagsNone* = DBusServerFlags({})
< proc none*(t: typedesc[DBusServerFlags]): DBusServerFlags = DBusServerFlags({})
< 
21081,21084d20980
< const
<   TlsDatabaseLookupFlagsNone* = TlsDatabaseLookupFlags({})
< proc none*(t: typedesc[TlsDatabaseLookupFlags]): TlsDatabaseLookupFlags = TlsDatabaseLookupFlags({})
< 
21311,21314d21206
< const
<   TlsPasswordFlagsNone* = TlsPasswordFlags({})
< proc none*(t: typedesc[TlsPasswordFlags]): TlsPasswordFlags = TlsPasswordFlags({})
< 
22279,22282d22170
< const
<   FileMeasureFlagsNone* = FileMeasureFlags({})
< proc none*(t: typedesc[FileMeasureFlags]): FileMeasureFlags = FileMeasureFlags({})
< 
22728,22731d22615
< const
<   IOModuleScopeFlagsNone* = IOModuleScopeFlags({})
< proc none*(t: typedesc[IOModuleScopeFlags]): IOModuleScopeFlags = IOModuleScopeFlags({})
< 
23437,23440d23320
< const
<   SubprocessFlagsNone* = SubprocessFlags({})
< proc none*(t: typedesc[SubprocessFlags]): SubprocessFlags = SubprocessFlags({})
< 
25600,25603d25479
< const
<   ResolverNameLookupFlagsDefault* = ResolverNameLookupFlags({})
< proc default*(t: typedesc[ResolverNameLookupFlags]): ResolverNameLookupFlags = ResolverNameLookupFlags({})
< 
25983,25986d25858
< const
<   ResourceFlagsNone* = ResourceFlags({})
< proc none*(t: typedesc[ResourceFlags]): ResourceFlags = ResourceFlags({})
< 
26976,26979d26847
< const
<   SettingsBindFlagsDefault* = SettingsBindFlags({})
< proc default*(t: typedesc[SettingsBindFlags]): SettingsBindFlags = SettingsBindFlags({})
< 
28220,28223d28087
< 
< const
<   SocketMsgFlagsNone* = SocketMsgFlags({})
< proc none*(t: typedesc[SocketMsgFlags]): SocketMsgFlags = SocketMsgFlags({})
diff nim_gi/glib.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/glib.nim
2727,2730d2726
< const
<   FileSetContentsFlagsNone* = FileSetContentsFlags({})
< proc none*(t: typedesc[FileSetContentsFlags]): FileSetContentsFlags = FileSetContentsFlags({})
< 
2751,2754d2746
< const
<   FormatSizeFlagsDefault* = FormatSizeFlags({})
< proc default*(t: typedesc[FormatSizeFlags]): FormatSizeFlags = FormatSizeFlags({})
< 
4320,4323d4311
< const
<   KeyFileFlagsNone* = KeyFileFlags({})
< proc none*(t: typedesc[KeyFileFlags]): KeyFileFlags = KeyFileFlags({})
< 
6522,6525d6509
< const
<   OptionFlagsNone* = OptionFlags({})
< proc none*(t: typedesc[OptionFlags]): OptionFlags = OptionFlags({})
< 
7445,7448d7428
< const
<   SpawnFlagsDefault* = SpawnFlags({})
< proc default*(t: typedesc[SpawnFlags]): SpawnFlags = SpawnFlags({})
< 
8821,8824d8800
< const
<   UriHideFlagsNone* = UriHideFlags({})
< proc none*(t: typedesc[UriHideFlags]): UriHideFlags = UriHideFlags({})
< 
8841,8844d8816
< const
<   UriParamsFlagsNone* = UriParamsFlags({})
< proc none*(t: typedesc[UriParamsFlags]): UriParamsFlags = UriParamsFlags({})
< 
8862,8865d8833
< 
< const
<   UriFlagsNone* = UriFlags({})
< proc none*(t: typedesc[UriFlags]): UriFlags = UriFlags({})
diff nim_gi/gobject.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gobject.nim
709,712d708
< const
<   BindingFlagsDefault* = BindingFlags({})
< proc default*(t: typedesc[BindingFlags]): BindingFlags = BindingFlags({})
< 
diff nim_gi/gstallocators.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gstallocators.nim
183,186d182
< const
<   FdMemoryFlagsNone* = FdMemoryFlags({})
< proc none*(t: typedesc[FdMemoryFlags]): FdMemoryFlags = FdMemoryFlags({})
< 
diff nim_gi/gstaudio.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gstaudio.nim
963,966d962
< const
<   AudioChannelMixerFlagsNone* = AudioChannelMixerFlags({})
< proc none*(t: typedesc[AudioChannelMixerFlags]): AudioChannelMixerFlags = AudioChannelMixerFlags({})
< 
1192,1195d1187
< const
<   AudioConverterFlagsNone* = AudioConverterFlags({})
< proc none*(t: typedesc[AudioConverterFlags]): AudioConverterFlags = AudioConverterFlags({})
< 
1908,1911d1899
< const
<   AudioFlagsNone* = AudioFlags({})
< proc none*(t: typedesc[AudioFlags]): AudioFlags = AudioFlags({})
< 
1928,1931d1915
< const
<   AudioPackFlagsNone* = AudioPackFlags({})
< proc none*(t: typedesc[AudioPackFlags]): AudioPackFlags = AudioPackFlags({})
< 
2017,2020d2000
< const
<   AudioQuantizeFlagsNone* = AudioQuantizeFlags({})
< proc none*(t: typedesc[AudioQuantizeFlags]): AudioQuantizeFlags = AudioQuantizeFlags({})
< 
2117,2120d2096
< 
< const
<   AudioResamplerFlagsNone* = AudioResamplerFlags({})
< proc none*(t: typedesc[AudioResamplerFlags]): AudioResamplerFlags = AudioResamplerFlags({})
diff nim_gi/gstbase.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gstbase.nim
755,758d754
< const
<   BaseParseFrameFlagsNone* = BaseParseFrameFlags({})
< proc none*(t: typedesc[BaseParseFrameFlags]): BaseParseFrameFlags = BaseParseFrameFlags({})
< 
diff nim_gi/gst.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gst.nim
1702,1705d1701
< const
<   StreamFlagsNone* = StreamFlags({})
< proc none*(t: typedesc[StreamFlags]): StreamFlags = StreamFlags({})
< 
3853,3856d3848
< const
<   BufferCopyFlagsNone* = BufferCopyFlags({})
< proc none*(t: typedesc[BufferCopyFlags]): BufferCopyFlags = BufferCopyFlags({})
< 
4419,4422d4410
< const
<   PluginDependencyFlagsNone* = PluginDependencyFlags({})
< proc none*(t: typedesc[PluginDependencyFlags]): PluginDependencyFlags = PluginDependencyFlags({})
< 
8562,8563d8549
<   SegmentFlagsNone* = SegmentFlags({})
< proc none*(t: typedesc[SegmentFlags]): SegmentFlags = SegmentFlags({})
10235,10238d10220
< const
<   BufferPoolAcquireFlagsNone* = BufferPoolAcquireFlags({})
< proc none*(t: typedesc[BufferPoolAcquireFlags]): BufferPoolAcquireFlags = BufferPoolAcquireFlags({})
< 
12183,12186d12164
< const
<   MetaFlagsNone* = MetaFlags({})
< proc none*(t: typedesc[MetaFlags]): MetaFlags = MetaFlags({})
< 
14992,14995d14969
< const
<   ParseFlagsNone* = ParseFlags({})
< proc none*(t: typedesc[ParseFlags]): ParseFlags = ParseFlags({})
< 
15698,15701d15671
< const
<   StackTraceFlagsNone* = StackTraceFlags({})
< proc none*(t: typedesc[StackTraceFlags]): StackTraceFlags = StackTraceFlags({})
< 
16505,16508d16474
< 
< const
<   TracerValueFlagsNone* = TracerValueFlags({})
< proc none*(t: typedesc[TracerValueFlags]): TracerValueFlags = TracerValueFlags({})
diff nim_gi/gstvideo.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gstvideo.nim
1184,1187d1183
< const
<   VideoChromaFlagsNone* = VideoChromaFlags({})
< proc none*(t: typedesc[VideoChromaFlags]): VideoChromaFlags = VideoChromaFlags({})
< 
1992,1995d1987
< const
<   VideoDitherFlagsNone* = VideoDitherFlags({})
< proc none*(t: typedesc[VideoDitherFlags]): VideoDitherFlags = VideoDitherFlags({})
< 
2264,2267d2255
< const
<   VideoFlagsNone* = VideoFlags({})
< proc none*(t: typedesc[VideoFlags]): VideoFlags = VideoFlags({})
< 
2289,2292d2276
< const
<   VideoPackFlagsNone* = VideoPackFlags({})
< proc none*(t: typedesc[VideoPackFlags]): VideoPackFlags = VideoPackFlags({})
< 
2509,2512d2492
< const
<   VideoMultiviewFlagsNone* = VideoMultiviewFlags({})
< proc none*(t: typedesc[VideoMultiviewFlags]): VideoMultiviewFlags = VideoMultiviewFlags({})
< 
2890,2893d2869
< const
<   VideoOverlayFormatFlagsNone* = VideoOverlayFormatFlags({})
< proc none*(t: typedesc[VideoOverlayFormatFlags]): VideoOverlayFormatFlags = VideoOverlayFormatFlags({})
< 
3071,3074d3046
< const
<   VideoResamplerFlagsNone* = VideoResamplerFlags({})
< proc none*(t: typedesc[VideoResamplerFlags]): VideoResamplerFlags = VideoResamplerFlags({})
< 
3164,3167d3135
< const
<   VideoScalerFlagsNone* = VideoScalerFlags({})
< proc none*(t: typedesc[VideoScalerFlags]): VideoScalerFlags = VideoScalerFlags({})
< 
3467,3470d3434
< 
< const
<   VideoTimeCodeFlagsNone* = VideoTimeCodeFlags({})
< proc none*(t: typedesc[VideoTimeCodeFlags]): VideoTimeCodeFlags = VideoTimeCodeFlags({})
diff nim_gi/gtk4.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gtk4.nim
216,219d215
< const
<   StateFlagsNormal* = StateFlags({})
< proc normal*(t: typedesc[StateFlags]): StateFlags = StateFlags({})
< 
36977,36980d36972
< const
<   StyleContextPrintFlagsNone* = StyleContextPrintFlags({})
< proc none*(t: typedesc[StyleContextPrintFlags]): StyleContextPrintFlags = StyleContextPrintFlags({})
< 
37389,37392d37380
< 
< const
<   PickFlagsDefault* = PickFlags({})
< proc default*(t: typedesc[PickFlags]): PickFlags = PickFlags({})
diff nim_gi/gtk.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gtk.nim
418,421d417
< const
<   StateFlagsNormal* = StateFlags({})
< proc normal*(t: typedesc[StateFlags]): StateFlags = StateFlags({})
< 
9033,9036d9028
< 
< const
<   StyleContextPrintFlagsNone* = StyleContextPrintFlags({})
< proc none*(t: typedesc[StyleContextPrintFlags]): StyleContextPrintFlags = StyleContextPrintFlags({})
diff nim_gi/gtksource5.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gtksource5.nim
858,861d857
< const
<   SortFlagsNone* = SortFlags({})
< proc none*(t: typedesc[SortFlags]): SortFlags = SortFlags({})
< 
4820,4823d4815
< 
< const
<   FileSaverFlagsNone* = FileSaverFlags({})
< proc none*(t: typedesc[FileSaverFlags]): FileSaverFlags = FileSaverFlags({})
diff nim_gi/gtksource.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/gtksource.nim
1021,1024d1020
< const
<   SortFlagsNone* = SortFlags({})
< proc none*(t: typedesc[SortFlags]): SortFlags = SortFlags({})
< 
4212,4215d4207
< 
< const
<   FileSaverFlagsNone* = FileSaverFlags({})
< proc none*(t: typedesc[FileSaverFlags]): FileSaverFlags = FileSaverFlags({})
diff nim_gi/pango.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/pango.nim
4372,4375d4371
< const
<   ShapeFlagsNone* = ShapeFlags({})
< proc none*(t: typedesc[ShapeFlags]): ShapeFlags = ShapeFlags({})
< 
4383,4386d4378
< 
< const
<   ShowFlagsNone* = ShowFlags({})
< proc none*(t: typedesc[ShowFlags]): ShowFlags = ShowFlags({})
diff nim_gi/rsvg.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/rsvg.nim
460,463d459
< const
<   HandleFlagsFlagsNone* = HandleFlags({})
< proc flagsNone*(t: typedesc[HandleFlags]): HandleFlags = HandleFlags({})
< 
diff nim_gi/vte.nim /home/salewski/.nimble/pkgs/gintro-#v0.9.2/gintro/vte.nim
298,301d297
< const
<   PtyFlagsDefault* = PtyFlags({})
< proc default*(t: typedesc[PtyFlags]): PtyFlags = PtyFlags({})
< 
```

**Comment by StefanSalewski** [comment]
Should be fixed in v0.9.4 now.
Will close this issue, please reopen if necessary.
